fil_fungible_token/src/token/mod.rs

```diff
@@ -433,337 +439,340 @@ mod test {
+        // total supply is unchanged
+        assert_eq!(token.total_supply(), TokenAmount::from(100));
+
+        // cannot transfer a negative value
+        token.transfer(ALICE, ALICE, BOB, &TokenAmount::from(-1), &[]).unwrap_err();
+        // balances are unchanged
+        assert_eq!(token.balance_of(ALICE).unwrap(), TokenAmount::from(40));
+        assert_eq!(token.balance_of(BOB).unwrap(), TokenAmount::from(60));
+        // total supply is unchanged
+        assert_eq!(token.total_supply(), TokenAmount::from(100));
+
+        // transfer zero value
+        token.transfer(ALICE, ALICE, BOB, &TokenAmount::zero(), &[]).unwrap();
+        // balances are unchanged
+        assert_eq!(token.balance_of(ALICE).unwrap(), TokenAmount::from(40));
+        assert_eq!(token.balance_of(BOB).unwrap(), TokenAmount::from(60));
+        // total supply is unchanged
+        assert_eq!(token.total_supply(), TokenAmount::from(100));
+
+        // transfer zero to self
+        token.transfer(ALICE, ALICE, ALICE, &TokenAmount::zero(), &[]).unwrap();
+        // balances are unchanged
+        assert_eq!(token.balance_of(ALICE).unwrap(), TokenAmount::from(40));
+        assert_eq!(token.balance_of(BOB).unwrap(), TokenAmount::from(60));
+        // total supply is unchanged
+        assert_eq!(token.total_supply(), TokenAmount::from(100));
+
+        // transfer value to self
+        token.transfer(ALICE, ALICE, ALICE, &TokenAmount::from(10), &[]).unwrap();
+        // balances are unchanged
+        assert_eq!(token.balance_of(ALICE).unwrap(), TokenAmount::from(40));
+        assert_eq!(token.balance_of(BOB).unwrap(), TokenAmount::from(60));
+        // total supply is unchanged
+        assert_eq!(token.total_supply(), TokenAmount::from(100));
     }
 
     #[test]
-    fn it_disallows_transfer_when_receiver_hook_aborts() {
+    fn it_fails_to_transfer_when_receiver_hook_aborts() {
         let mut token = new_token();
 
-        let owner = 1;
-        let receiver = 2;
         // mint 100 for owner
-        token.mint(TOKEN_ACTOR_ADDRESS, owner, &TokenAmount::from(100), &[]).unwrap();
+        token.mint(TOKEN_ACTOR_ADDRESS, ALICE, &TokenAmount::from(100), &[]).unwrap();
         // transfer 60 from owner -> receiver, but simulate receiver aborting the hook
-        token
-            .transfer(owner, owner, receiver, &TokenAmount::from(60), "abort".as_bytes())
-            .unwrap_err();
+        token.transfer(ALICE, ALICE, BOB, &TokenAmount::from(60), "abort".as_bytes()).unwrap_err();
 
-        // balances didn't change
-        let balance = token.balance_of(owner).unwrap();
-        assert_eq!(balance, TokenAmount::from(100));
-        let balance = token.balance_of(receiver).unwrap();
-        assert_eq!(balance, TokenAmount::from(0));
+        // balances unchanged
+        assert_eq!(token.balance_of(ALICE).unwrap(), TokenAmount::from(100));
+        assert_eq!(token.balance_of(BOB).unwrap(), TokenAmount::from(0));
```

Comment:
Just 'cause I'm paranoid, test transferring non-zero value to self with a receiver hook that aborts.